pkg/initializer/validator/datavalidator.go

```diff
@@ -79,6 +78,10 @@ func (d *DataValidator) Validate(mode Mode, failBelowRevision int64) (DataDirSta
 	if mode == Full {
 		d.Logger.Info("Checking for data directory files corruption...")
 		if err := d.checkForDataCorruption(); err != nil {
+			if err.Error() == ErrTimeout.Error() {
```

Comment:
This is just a string comparison, isn't it?
would `err.Error()` return `"timeout"` in this case?

Reply:
yes, it is just a string comparison.
Initially I tried to use [errors.Is()](https://pkg.go.dev/errors#Is) and [errors.As()](https://pkg.go.dev/errors#As) but they didn’t work as expected, so I used this.

Reply:
I guess my question here is, will err return an error with just the string `"timeout"`? 
that's the only way this condition evaluates to true, isn't it?

Reply:
Actually what you have defined is a `Sentinal` error and direct comparison with errors.Is(err, ErrTimeout) should work.
The issue is that you have now defined a duplicate error with the same name `ErrTimeout` that is also defined in bbolt/errors.go. So when bbolt returns a sentinal error `ErrTimeout` from `verifyDb` then you compare that against the sentinal error that you have defined in your file. This will never work.

If you wish to have your own error then in `verifyDb` method you should check if the error returned by `bolt.Open` is `ErrTimeout` as defined in `bolt` package and if yes then instead of propagating that error you wrap that error into a custom error and then return that instead.

Reply:
thanks @unmarshall ... I will update